server/src/utils/operatingSystem.ts

```diff
@@ -17,6 +17,61 @@ export async function runCmd(cmd: string, cwd?: string): Promise<string> {
   });
 }
 
+/**
+ * Executes a command using exec, writes provided input to stdin,
+ * and returns a Promise that resolves with stdout and stderr.
+ *
+ * @param {string} command - The command to execute.
+ * @param {string} input - The input string to write to the child process's stdin.
+ * @param {ExecOptions} options - The options to pass to the exec function.
+ * @param {number} timeout - The timeout in milliseconds.
+ * @returns {Promise<{stdout: string, stderr: string}>}
+ */
+export async function execWithInput(
+  command: string,
+  input: string,
+  options: ExecOptions,
+  timeout?: number
+): Promise<{ stdout: string; stderr: string }> {
+  return new Promise((resolve, reject) => {
+    // Set a timeout if provided
+    if (timeout !== undefined) {
+      setTimeout(() => {
+        reject("Timed out");
+      }, timeout);
+    }
+
+    // Start the child process
+    const child = exec(command, options, (error, stdout, stderr) => {
+      if (error) {
+        reject(error);
```

Comment:
I think we need to change this to return immediately. I am not sure how node will handle rejecting then continuing on to resolve.